clang/test/CIR/CodeGen/optnone.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: %clang_cc1 -O0 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s -check-prefix=CIR-O0
+// RUN: %clang_cc1 -O0 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=LLVM-O0
+
+// RUN: %clang_cc1 -O2 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t2.cir
+// RUN: FileCheck --input-file=%t2.cir %s -check-prefix=CIR-O2
+// RUN: %clang_cc1 -O2 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-llvm %s -o %t2.ll
+// RUN: FileCheck --input-file=%t2.ll %s -check-prefix=LLVM-O2
+
+int s0(int a, int b) {
+  int x = a + b;
+  if (x > 0)
+    x = 0;
+  else
+    x = 1;
+  return x;
+}
+
+// CIR-O0:   cir.func @_Z2s0ii(%arg0:{{.*}}, %arg1:{{.*}} -> {{.*}} extra( {inline = #cir.inline<no>, optnone = #cir.optnone} )
```

Comment:
I see the output is:

`extra( {inline = #cir.inline<no>, optnone = #cir.optnone} )`

Given that the idea is to be unit-attr-like, it seems too verbose to have `optnone = #cir.optnone`, can we instead have (this builds on the assumption that nothing will show up if this CIRUnitAttr doesn't get created, which is how unit-attr usually works):

`extra( {inline = #cir.inline<no>, #cir.optnone} )`

?

Reply:
Talked offline. It looks like MLIR has a hack that only works for the MLIR UnitAttr  https://github.com/llvm/clangir/blob/8e37695674b951d35d03945d4d30f19f14e4f164/mlir/lib/IR/AsmPrinter.cpp#L2584.   I can think to create a subclass to override that method (will need to change it to virtual first). The parser (https://github.com/llvm/clangir/blob/8e37695674b951d35d03945d4d30f19f14e4f164/mlir/lib/AsmParser/AttributeParser.cpp#L323)  will need to be overriden too. Doesn’t sound trivial.   The problem is that extra func attr is a list of named attributes, so no direct way to control how a CIR attribute is printed there. The name will always be printed first.